evm-validate/src/lib.rs

```diff
@@ -417,3 +468,764 @@ fn validate_transaction_hashes(
 
     Ok(())
 }
+
+// AccessListWrapper is a wrapper around the AccessList type that allows it to be deserialized from a byte array.
+#[derive(Deserialize)]
+pub struct AccessListWrapper(pub Vec<AccessListItemWrapper>);
+
+impl Into<AccessList> for AccessListWrapper {
+    fn into(self) -> AccessList {
+        AccessList(self.0.into_iter().map(|item| item.into()).collect())
+    }
+}
+
+// AccessListItemWrapper is a wrapper around the AccessListItem type that allows it to be deserialized from a byte array.
+#[derive(Deserialize)]
+pub struct AccessListItemWrapper {
+    pub address: Address,
+    pub storage_keys: Vec<B256>,
+}
+
+impl Into<AccessListItem> for AccessListItemWrapper {
+    fn into(self) -> AccessListItem {
+        AccessListItem {
+            address: self.address,
+            storage_keys: self.storage_keys,
+        }
+    }
+}
+
+pub fn validate_root_hashes(
+    blocks: &RecordBatch,
+    logs: &RecordBatch,
+    transactions: &RecordBatch,
+) -> Result<()> {
+
+    // CREATE A LOG MAPPING
+
+    let log_array = extract_log_cols_as_arrays(logs)?;
+
+    // get first log block num and tx idx
+    let mut current_block_num = log_array.block_number.value(0);
+    let mut current_tx_idx = log_array.log_index.value(0);
+    // initialize a vec to store all logs for a tx
+    let mut tx_logs = Vec::<Log>::with_capacity(20);
+    // initialize a map to store logs by block num and tx idx   
+    let mut logs_by_block_num_and_tx_idx = BTreeMap::<(u64, u64), Vec<Log>>::new();
+
+    let log_iterators = log_array.block_number.iter()
```

Comment:
can use
 
```rust
for i in 0..block_number.len() {
   let bn = block_number.get_value(i);
   let hash = hash.get_value(i);
}
```

this pattern produces less code and less tuple (de)construction

Reply:
I've tried using a for loop to test. When you use iter, it returns an Option if the value is empty, which is good for many fields. I already went and implemented this approach; I don't want to change it.